### PR TITLE
Add TiTiler service and update to allow passing ID in the url path

### LINKED ISSUE #1: Merge in TiTiler deployment stack
The TiTiler deployment was create with an older version of CDK and needs some updating before it can be fully integrated with this repo.

### PR DESCRIPTION
**Pull request by eliotjordan**
This PR:

1. Ports TiTiler from an older CDK repo.
2. Moves any rewrite logic from a "sidecar" lambda to a python middleware.
3. Moves passing of the resource ID from a query param to the path of the URL.
4. Decouples Figgy from tile serving logic.
5. Ensures that TileJSON documents are generated with the correct tile url.

Closes #1

### CODE REVIEW

File: resources/middleware.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        scope["headers"] = [(k, v) for k, v in headers.items()]
+
+        await self.app(scope, receive, send)
+
+# Middleware to rewrite requests URLs before passing to TiTiler.
+# https://map-tiles.princeton.edu/{id}/{service} ->
+# https://map-tiles.princeton.edu/{service}?url=s3://{bucket}/{path}
+#
+# Examples:
+#   https://map-tiles.princeton.edu/1234567/mosaicjson?
```
Comment (hackartisan) [inline resources/middleware.py]: delete question mark at the end?

File: resources/middleware.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+#   https://map-tiles.princeton.edu/mosaicjson?url=s3://figgy-geo-production/12/34/56/1234567/mosaic.json
+#
+#   https://map-tiles.princeton.edu/1234567/cog?id=1234567 ->
```
Comment (hackartisan) [inline resources/middleware.py]: delete this query string at the end of this?

**Review by hackartisan — CHANGES_REQUESTED** [review]
A couple of typos in comments I think. Also we should add to the README about how to deploy the titiler piece.

File: resources/middleware.py (hunk 1 above)
Comment (eliotjordan) [reply resources/middleware.py]: Done

File: resources/middleware.py (hunk 2 above)
Comment (eliotjordan) [reply resources/middleware.py]: Done.

File: geoservices/titiler_service_stack.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                    command=["bash", "-c", "cp -R /var/task/. /asset-output/."],
+                ),
+            ),
+            handler="handler.handler",
+            memory_size=3008,
+            timeout=Duration.seconds(600),
+            environment=env,
+        )
+
+        # TODO: Remove when staging and production no longer use lambda
```
Comment (hackartisan) [inline geoservices/titiler_service_stack.py]: For these 2 TODOs: How can I assess whether they are done? Is there a ticket we can link to or some documentation or change log?

**Review by hackartisan — CHANGES_REQUESTED** [review]
One more documentation question

File: geoservices/titiler_service_stack.py (hunk 3 above)
Comment (eliotjordan) [reply geoservices/titiler_service_stack.py]: Those were vestigal and not used. We are using a rewrite middleware instead of a lambda now. Removed.

**Review by hackartisan — APPROVED** [review]

**Pull request merged**